core/common/src/main/kotlin/eu/kanade/tachiyomi/util/storage/SevenZUtil.kt

```diff
@@ -0,0 +1,35 @@
+package eu.kanade.tachiyomi.util.storage
+
+import eu.kanade.tachiyomi.util.lang.compareToCaseInsensitiveNaturalOrder
+import org.apache.commons.compress.archivers.dump.UnsupportedCompressionAlgorithmException
+import org.apache.commons.compress.archivers.sevenz.SevenZFile
+import org.apache.commons.compress.archivers.sevenz.SevenZMethod
+import tachiyomi.core.common.util.system.ImageUtil
+import java.io.IOException
+import java.io.InputStream
+
+object SevenZUtil {
+    private val GoodMethods = arrayOf(
+        SevenZMethod.LZMA2,
+        SevenZMethod.DEFLATE,
+        SevenZMethod.COPY,
+    )
+
+    fun SevenZFile.getImages(notifySlowArchives: (method: String) -> Unit): Sequence<ByteArray> {
+        return generateSequence {
+            try {
+                getNextEntry()
+            } catch (e: IOException) {
+                throw UnsupportedCompressionAlgorithmException()
```

Comment:
Exception seems to come from `dump` archive. Make a new exception for unsupported 7zip archive one I suppose?